packages/api-key/src/services/api-key-module-service.ts

```diff
@@ -65,28 +72,60 @@ export default class ApiKeyModuleService<TEntity extends ApiKey = ApiKey>
     data: ApiKeyTypes.CreateApiKeyDTO | ApiKeyTypes.CreateApiKeyDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<ApiKeyTypes.ApiKeyDTO | ApiKeyTypes.ApiKeyDTO[]> {
-    const createdApiKeys = await this.create_(data, sharedContext)
+    const [createdApiKeys, generatedTokens] = await this.create_(
+      Array.isArray(data) ? data : [data],
+      sharedContext
+    )
 
-    return await this.baseRepository_.serialize<
-      ApiKeyTypes.ApiKeyDTO | ApiKeyTypes.ApiKeyDTO[]
+    const serializedResponse = await this.baseRepository_.serialize<
+      ApiKeyTypes.ApiKeyDTO[]
     >(createdApiKeys, {
       populate: true,
     })
+
+    // When creating we want to return the raw token, as this will be the only time the user will be able to take note of it for future use.
+    const responseWithRawToken = serializedResponse.map((key) => ({
+      ...key,
+      token:
+        generatedTokens.find((t) => t.hashedToken === key.token)?.rawToken ??
+        key.token,
+    }))
+
+    return Array.isArray(data) ? responseWithRawToken : responseWithRawToken[0]
   }
 
   @InjectTransactionManager("baseRepository_")
   protected async create_(
-    data: ApiKeyTypes.CreateApiKeyDTO | ApiKeyTypes.CreateApiKeyDTO[],
+    data: ApiKeyTypes.CreateApiKeyDTO[],
     @MedusaContext() sharedContext: Context = {}
-  ): Promise<TEntity | TEntity[]> {
-    const data_ = Array.isArray(data) ? data : [data]
+  ): Promise<[TEntity[], TokenDTO[]]> {
+    await this.validateCreateApiKeys(data, sharedContext)
```

Comment:
suggestion: I think we have a convention of calling protected methods with a `_` in the end
```suggestion
    await this.validateCreateApiKeys_(data, sharedContext)
```

Reply:
Fine with that if we prefer it 👍